src/vs/workbench/api/common/extHostNotebookDocument.ts

```diff
@@ -114,12 +114,16 @@ export class ExtHostCell {
 
 	setOutputItems(outputId: string, append: boolean, newOutputItems: extHostProtocol.NotebookOutputItemDto[]) {
 		const newItems = newOutputItems.map(extHostTypeConverters.NotebookCellOutputItem.to);
-		const output = this._outputs.find(op => op.id === outputId);
-		if (output) {
+
+		// Use an index so we can overwrite internal properties (otherwise readonly)
+		const index = this._outputs.findIndex(op => op.id === outputId);
+		if (index >= 0) {
+			const output = deepClone(this._outputs[index]);
 			if (!append) {
-				output.items.length = 0;
+				output.items.length = 0; // This crashes without a copy
```

Comment:
I also noticed this throwing an exception while running the same code.

Reply:
You have a stack trace? Is it throwing because the event processing froze it? That wouldn't be good...

Reply:
Yes I believe it's crashing because of the freeze.